app/controllers/solidus_subscriptions/api/v1/subscriptions_controller.rb

```diff
@@ -6,6 +6,18 @@ module V1
       class SubscriptionsController < BaseController
         protect_from_forgery unless: -> { request.format.json? }
 
+        def create
+          store = params[:store_id].nil? ? ::Spree::Store.default : ::Spree::Store.find(id: params[:store_id])
+          attributes = subscription_params.merge(user: current_api_user, store: store)
```

Comment:
I think that [the default subscription_attributes](https://github.com/solidusio-contrib/solidus_subscriptions/blob/master/lib/solidus_subscriptions/configuration.rb#L63-L73) aren't enough for a `create` endpoint. For example, I don't see any way to specify a payment method 🤔

Reply:
ATM, this is working:
```ruby
SolidusSubscriptions::Subscription.new(store: Spree::Store.default, user: Spree::User.last, interval_length: 1).validate!
  Spree::Store Load (1.7ms)  SELECT "spree_stores".* FROM "spree_stores" WHERE "spree_stores"."default" = $1 ORDER BY "spree_stores"."id" ASC LIMIT $2  [["default", true], ["LIMIT", 1]]
  Spree::User Load (0.3ms)  SELECT "spree_users".* FROM "spree_users" WHERE "spree_users"."deleted_at" IS NULL ORDER BY "spree_users"."id" DESC LIMIT $1  [["LIMIT", 1]]
=> true
```

Reply:
I would expect the validation to fails if the payment method (or something else) is necessary while creating the Subscription.

Reply:
how can I create a subscription with a specific payment method using the API? Maybe I'm missing something

Reply:
My knowledge of this extension is limited too 😄 
From what I see here: https://github.com/solidusio-contrib/solidus_subscriptions/blob/master/app/models/solidus_subscriptions/subscription.rb#L21
the payment method is optional.

I see your point but the requirement of the issue doesn't talk about setting the payment method via API.

@aldesantis: what do you think about this point?
I could simply permit payment_method / payment_source attrs if we need them.

Reply:
@blocknotes if you don't specify the payment method method, payment source, billing or shipping address, the extension will pick the defaults from your wallet/address book (see [here](https://github.com/solidusio-contrib/solidus_subscriptions/blob/master/app/models/solidus_subscriptions/subscription.rb#L223)).

With that said, I think we should permit `payment_method`/`payment_source` in order to expose the full functionality of the extension to the end users.

Reply:
@aldesantis, @ChristianRimondi: I added a new commit to handle the payment attributes. I also included the address ids because it looks useful to me: https://github.com/solidusio-contrib/solidus_subscriptions/pull/199/commits/88383e57bd20d158c486d44471a465b291c7b10c

Can you go on with the review?